Improve sentences
----------------------------

In this notebook we want to improve the corpora of new sentences and the final corpora which is the concatenation of the original corpora and the new one (we got the two in the following notebook [new_sentences](extract_new_sentences.ipynb)). 

For the improvement we want to make the following modifications that we found to be necessary after fine-tuning the `t5` model on the new sentences:

- Remove the definitions: We had already extract the new sentences without the definitions we will only need to load the "true sentences".
- Replace exclamation marks (!) at the end of the phrases by period marks (.): This will help the transformers to identify the end of a sentence. Not that we will do this for both of the new sentences and the last version corpora (v4). We cannot know if it is the right thing to do in order to obtain better performance that's why we will create a function and add it when making hyper-parameter search and reduce the search space if necessary.
- Remove sentences with less than 3 tokens in one of the French of Wolof version. This will reduce the number of sentences but can provide better agreement with the right way to write sentences.

The above modifications on the version 4 of the corpora and the diagne's sentences will be necessary to create the version 5 of the corpora. The additional version of the diagne's sentences and 6 of the corpora will understand additionally sentences got from `omniglot`. We will not select the sentences by length for the new versions.

All the modifications can be done with the `pandas` library.

In [2]:
import pandas as pd
from tokenizers.pre_tokenizers import Whitespace
import re

### Remove the definitions

Let us load the two corpora.

In [3]:
corpora_v4 = pd.read_csv('data/extractions/new_data/corpora_v4.csv')

diagne_sentences = pd.read_csv('data/extractions/new_data/sentences.csv')

In [4]:
corpora_v4.shape[0], diagne_sentences.shape[0]

(2500, 1615)

### Add periods to the end of the sentences

We have three possibilities:

- We can replace the exclamations by period marks (option 2)
- ... add period mark at each end of sentences (option 3)
- ... do the two previous modification at same time (option 4)

The following function will be added inside the `TransformerSequences` class when recuperating the sentences before the training.

In [5]:
%%writefile wolof-translate/wolof_translate/utils/improvements/end_marks.py

import pandas as pd
from typing import *

def add_end_mark(sentences: Union[list, str], end_mark: str = ".", end_mark_to_remove: Union[str, None] = None, 
                 replace: bool = False, poss_end_marks: list = ['!', '?', '.', '...']):
    
    if isinstance(sentences, str): sentences = [sentences]
    
    if replace and end_mark_to_remove is None:
        
        raise ValueError("You must provide a end mark to remove if you want to make replacement !")
    
    new_sentences = []
    
    # if replace is chosen we will replace the end to remove by the end mark if not only add end mark to each sentence and remove the end mark to remove
    for sentence in sentences:
        
        if replace:
            
            if sentence[-1] == end_mark_to_remove:
                
                sentence = sentence[:-1].strip() + end_mark
            
        else:
            
            if not end_mark_to_remove is None and sentence[-1] == end_mark_to_remove:
                
                sentence = sentence[:-1]
            
            if not sentence[-1] in poss_end_marks: 
                
                sentence = sentence.strip() + end_mark
        
        new_sentences.append(sentence)
    
    return new_sentences
    

Overwriting wolof-translate/wolof_translate/utils/improvements/end_marks.py


Let us try the different type of end mark adding with the above function.

In [6]:
%run wolof-translate/wolof_translate/utils/improvements/end_marks.py

In [6]:
# recuperate some sentences from the diagne's sentences
sample = diagne_sentences.sample(30)

In [7]:
sample

french  \
1434  Je te confie celui-là dont il fut question au ...   
381                                         L'homme ira   
284                              C'est quand tu es venu   
704                                   À qui parles-tu ?   
1156                   J'ai été, tu as été, il a été...   
1359                Cet homme est Laobe de Saint-Louis.   
1365                        C'est moi qui dois partir !   
1234         Les enfants qui venaient ne viennent plus.   
976                                       Me voici... !   
1508       C'était ceux-là même jusqu'à ce qu'il parte.   
885                       J'ai vu les enfants sauf toi.   
848                       Il a tant parlé que le voilà.   
492                                         Je partirai   
999                 Je n'avais été nulle part ce matin.   
403                            C'est moi le professeur.   
707                                Tu pries pour quoi ?   
456                                Tu ne viendras pas ?   
1304                 C'est d'autres gens que j'ai vus !   
1466               Il lui a amené du fourrage au champ.   
310                           L'homme qui eût travaillé   
1059                            C'est toi qu'elle veut.   
710                                         C'est bon ?   
1550                 Tu ne vas pas dans un autre lieu !   
107                          Va voir tous les chevaux !   
644                                     Tu as dit cela.   
887                         Parle au sujet de ceux-ci !   
200                              Tu vois ce garçon-là ?   
1080               J'ai été dans une maison spacieuse !   
1172                                        Oui Samba !   
1310                        Tu as vu quels amis à toi ?   

                              wolof  
1434      Deŋk naa la boobule woon.  
381               Góor gi dana demi  
284                 Bi ŋga ñëwée la  
704                    Kooy waxal ?  
1156    Dem naa, dem ŋga, dem na...  
1359         Gor gii di Lawbe Ndar.  
1365                   Maa di dem !  
1234  Xale yooya daan ñëw ñëwëtuñu.  
976                   Maa ŋgii... !  
1508   Ñawoon la ba bi mu fiy joge.  
885    Gis naa xale yi gannaaw yaw.  
848               Wax na ba mi ŋgi.  
492                       Dinaa dem  
999    Demumawoon fenn tày ci suba.  
403           Maa di jaŋgalékat bi.  
707                     Looy ñaan ?  
456                       Doo ñëw ?  
1304           Yeneen nit laa gis !  
1466  Yobul na ka ŋgoon ca tool ya.  
310              Waa ji liggéeykoon  
1059                   Yaw la bëgg.  
710                       Neex na ?  
1550               Demuloo feneen !  
107               Seetil fas yépp !  
644                  La ŋga wax la.  
887                  Waxal ci ñii !  
200              Gis ŋga xale bii ?  
1080        Dem naa ci kër gu yaa !  
1172                   Waaw Samba !  
1310         Gis ŋga sa yan xarit ?

We need to add the modification on each version of the sentences.

- Replace exclamations by end marks.

In [8]:
french_sents = sample['french'].to_list()

wolof_sents = sample['wolof'].to_list()

french_sents = add_end_mark(french_sents, end_mark_to_remove='!', replace=True)

wolof_sents = add_end_mark(wolof_sents, end_mark_to_remove='!', replace=True)

pd.DataFrame({'french': french_sents, 'wolof': wolof_sents})


french  \
0   Je te confie celui-là dont il fut question au ...   
1                                         L'homme ira   
2                              C'est quand tu es venu   
3                                   À qui parles-tu ?   
4                    J'ai été, tu as été, il a été...   
5                 Cet homme est Laobe de Saint-Louis.   
6                          C'est moi qui dois partir.   
7          Les enfants qui venaient ne viennent plus.   
8                                        Me voici....   
9        C'était ceux-là même jusqu'à ce qu'il parte.   
10                      J'ai vu les enfants sauf toi.   
11                      Il a tant parlé que le voilà.   
12                                        Je partirai   
13                Je n'avais été nulle part ce matin.   
14                           C'est moi le professeur.   
15                               Tu pries pour quoi ?   
16                               Tu ne viendras pas ?   
17                  C'est d'autres gens que j'ai vus.   
18               Il lui a amené du fourrage au champ.   
19                          L'homme qui eût travaillé   
20                            C'est toi qu'elle veut.   
21                                        C'est bon ?   
22                  Tu ne vas pas dans un autre lieu.   
23                          Va voir tous les chevaux.   
24                                    Tu as dit cela.   
25                         Parle au sujet de ceux-ci.   
26                             Tu vois ce garçon-là ?   
27                J'ai été dans une maison spacieuse.   
28                                         Oui Samba.   
29                        Tu as vu quels amis à toi ?   

                            wolof  
0       Deŋk naa la boobule woon.  
1               Góor gi dana demi  
2                 Bi ŋga ñëwée la  
3                    Kooy waxal ?  
4     Dem naa, dem ŋga, dem na...  
5          Gor gii di Lawbe Ndar.  
6                     Maa di dem.  
7   Xale yooya daan ñëw ñëwëtuñu.  
8                    Maa ŋgii....  
9    Ñawoon la ba bi mu fiy joge.  
10   Gis naa xale yi gannaaw yaw.  
11              Wax na ba mi ŋgi.  
12                      Dinaa dem  
13   Demumawoon fenn tày ci suba.  
14          Maa di jaŋgalékat bi.  
15                    Looy ñaan ?  
16                      Doo ñëw ?  
17            Yeneen nit laa gis.  
18  Yobul na ka ŋgoon ca tool ya.  
19             Waa ji liggéeykoon  
20                   Yaw la bëgg.  
21                      Neex na ?  
22                Demuloo feneen.  
23               Seetil fas yépp.  
24                 La ŋga wax la.  
25                  Waxal ci ñii.  
26             Gis ŋga xale bii ?  
27         Dem naa ci kër gu yaa.  
28                    Waaw Samba.  
29         Gis ŋga sa yan xarit ?

- Add end marks at each sentence without replacing or removing

In [9]:
french_sents = sample['french'].to_list()

wolof_sents = sample['wolof'].to_list()

french_sents = add_end_mark(french_sents)

wolof_sents = add_end_mark(wolof_sents)

pd.DataFrame({'french': french_sents, 'wolof': wolof_sents})


french  \
0   Je te confie celui-là dont il fut question au ...   
1                                        L'homme ira.   
2                             C'est quand tu es venu.   
3                                   À qui parles-tu ?   
4                    J'ai été, tu as été, il a été...   
5                 Cet homme est Laobe de Saint-Louis.   
6                         C'est moi qui dois partir !   
7          Les enfants qui venaient ne viennent plus.   
8                                       Me voici... !   
9        C'était ceux-là même jusqu'à ce qu'il parte.   
10                      J'ai vu les enfants sauf toi.   
11                      Il a tant parlé que le voilà.   
12                                       Je partirai.   
13                Je n'avais été nulle part ce matin.   
14                           C'est moi le professeur.   
15                               Tu pries pour quoi ?   
16                               Tu ne viendras pas ?   
17                 C'est d'autres gens que j'ai vus !   
18               Il lui a amené du fourrage au champ.   
19                         L'homme qui eût travaillé.   
20                            C'est toi qu'elle veut.   
21                                        C'est bon ?   
22                 Tu ne vas pas dans un autre lieu !   
23                         Va voir tous les chevaux !   
24                                    Tu as dit cela.   
25                        Parle au sujet de ceux-ci !   
26                             Tu vois ce garçon-là ?   
27               J'ai été dans une maison spacieuse !   
28                                        Oui Samba !   
29                        Tu as vu quels amis à toi ?   

                            wolof  
0       Deŋk naa la boobule woon.  
1              Góor gi dana demi.  
2                Bi ŋga ñëwée la.  
3                    Kooy waxal ?  
4     Dem naa, dem ŋga, dem na...  
5          Gor gii di Lawbe Ndar.  
6                    Maa di dem !  
7   Xale yooya daan ñëw ñëwëtuñu.  
8                   Maa ŋgii... !  
9    Ñawoon la ba bi mu fiy joge.  
10   Gis naa xale yi gannaaw yaw.  
11              Wax na ba mi ŋgi.  
12                     Dinaa dem.  
13   Demumawoon fenn tày ci suba.  
14          Maa di jaŋgalékat bi.  
15                    Looy ñaan ?  
16                      Doo ñëw ?  
17           Yeneen nit laa gis !  
18  Yobul na ka ŋgoon ca tool ya.  
19            Waa ji liggéeykoon.  
20                   Yaw la bëgg.  
21                      Neex na ?  
22               Demuloo feneen !  
23              Seetil fas yépp !  
24                 La ŋga wax la.  
25                 Waxal ci ñii !  
26             Gis ŋga xale bii ?  
27        Dem naa ci kër gu yaa !  
28                   Waaw Samba !  
29         Gis ŋga sa yan xarit ?

- Remove exclamation marks and add period marks

In [10]:
french_sents = sample['french'].to_list()

wolof_sents = sample['wolof'].to_list()

french_sents = add_end_mark(french_sents, end_mark_to_remove='!')

wolof_sents = add_end_mark(wolof_sents, end_mark_to_remove='!')

pd.DataFrame({'french': french_sents, 'wolof': wolof_sents})


french  \
0   Je te confie celui-là dont il fut question au ...   
1                                        L'homme ira.   
2                             C'est quand tu es venu.   
3                                   À qui parles-tu ?   
4                    J'ai été, tu as été, il a été...   
5                 Cet homme est Laobe de Saint-Louis.   
6                          C'est moi qui dois partir.   
7          Les enfants qui venaient ne viennent plus.   
8                                        Me voici....   
9        C'était ceux-là même jusqu'à ce qu'il parte.   
10                      J'ai vu les enfants sauf toi.   
11                      Il a tant parlé que le voilà.   
12                                       Je partirai.   
13                Je n'avais été nulle part ce matin.   
14                           C'est moi le professeur.   
15                               Tu pries pour quoi ?   
16                               Tu ne viendras pas ?   
17                  C'est d'autres gens que j'ai vus.   
18               Il lui a amené du fourrage au champ.   
19                         L'homme qui eût travaillé.   
20                            C'est toi qu'elle veut.   
21                                        C'est bon ?   
22                  Tu ne vas pas dans un autre lieu.   
23                          Va voir tous les chevaux.   
24                                    Tu as dit cela.   
25                         Parle au sujet de ceux-ci.   
26                             Tu vois ce garçon-là ?   
27                J'ai été dans une maison spacieuse.   
28                                         Oui Samba.   
29                        Tu as vu quels amis à toi ?   

                            wolof  
0       Deŋk naa la boobule woon.  
1              Góor gi dana demi.  
2                Bi ŋga ñëwée la.  
3                    Kooy waxal ?  
4     Dem naa, dem ŋga, dem na...  
5          Gor gii di Lawbe Ndar.  
6                     Maa di dem.  
7   Xale yooya daan ñëw ñëwëtuñu.  
8                    Maa ŋgii....  
9    Ñawoon la ba bi mu fiy joge.  
10   Gis naa xale yi gannaaw yaw.  
11              Wax na ba mi ŋgi.  
12                     Dinaa dem.  
13   Demumawoon fenn tày ci suba.  
14          Maa di jaŋgalékat bi.  
15                    Looy ñaan ?  
16                      Doo ñëw ?  
17            Yeneen nit laa gis.  
18  Yobul na ka ŋgoon ca tool ya.  
19            Waa ji liggéeykoon.  
20                   Yaw la bëgg.  
21                      Neex na ?  
22                Demuloo feneen.  
23               Seetil fas yépp.  
24                 La ŋga wax la.  
25                  Waxal ci ñii.  
26             Gis ŋga xale bii ?  
27         Dem naa ci kër gu yaa.  
28                    Waaw Samba.  
29         Gis ŋga sa yan xarit ?

### Remove sentences with less than 3 tokens

Only the words interest us and we want to separate them by space or marks. Notice also that we want verify for each version and if one it is not accepted then the another one also.

In [11]:
def select_by_length(sentences: pd.DataFrame, min_len: int = 4):
    
    new_sentences = {'french': [], 'wolof': []}
    
    for i in list(sentences.index):
        
        sent1 = sentences.loc[i, 'french']
        
        sent2 = sentences.loc[i, 'wolof']
        
        tokens1 = re.findall('\w+', sent1)
        
        tokens2 = re.findall('\w+', sent2)
        
        take = True
        
        if len(tokens1) < min_len or len(tokens2) < min_len:
            
            take = False
        
        if take:
            
            new_sentences['french'].append(sent1)
            
            new_sentences['wolof'].append(sent2)
            
    return pd.DataFrame(new_sentences)

Let us make an example with the sample that we got earlier.

In [12]:
sample = select_by_length(sample, min_len=4)


In [13]:
sample

french  \
0   Je te confie celui-là dont il fut question au ...   
1                              C'est quand tu es venu   
2                    J'ai été, tu as été, il a été...   
3                 Cet homme est Laobe de Saint-Louis.   
4          Les enfants qui venaient ne viennent plus.   
5        C'était ceux-là même jusqu'à ce qu'il parte.   
6                       J'ai vu les enfants sauf toi.   
7                       Il a tant parlé que le voilà.   
8                 Je n'avais été nulle part ce matin.   
9                            C'est moi le professeur.   
10                 C'est d'autres gens que j'ai vus !   
11               Il lui a amené du fourrage au champ.   
12                                    Tu as dit cela.   
13                             Tu vois ce garçon-là ?   
14               J'ai été dans une maison spacieuse !   
15                        Tu as vu quels amis à toi ?   

                            wolof  
0       Deŋk naa la boobule woon.  
1                 Bi ŋga ñëwée la  
2     Dem naa, dem ŋga, dem na...  
3          Gor gii di Lawbe Ndar.  
4   Xale yooya daan ñëw ñëwëtuñu.  
5    Ñawoon la ba bi mu fiy joge.  
6    Gis naa xale yi gannaaw yaw.  
7               Wax na ba mi ŋgi.  
8    Demumawoon fenn tày ci suba.  
9           Maa di jaŋgalékat bi.  
10           Yeneen nit laa gis !  
11  Yobul na ka ŋgoon ca tool ya.  
12                 La ŋga wax la.  
13             Gis ŋga xale bii ?  
14        Dem naa ci kër gu yaa !  
15         Gis ŋga sa yan xarit ?

It only kept 18 samples from 20 ones. Let us make the same for all the sentences.

In [14]:
diagne_sentences = select_by_length(diagne_sentences)

corpora_v4 = select_by_length(corpora_v4)

Let us verify how many sentences we have kept.

In [15]:
# for diagne's sentences
diagne_sentences.shape[0]

970

In [16]:
# in corpora v4
corpora_v4.shape[0]

1849

We removed $651$ sentences from the corpora v4 and $645$ sentences from the diagne's sentences.

Let us save the results to make again hyperparameter search with them.

In [17]:
# save the new version of the diagne's sentences in red_sentences.csv
diagne_sentences.to_csv('data/extractions/new_data/red_sentences.csv', index = False)

# save the new version of the corpora
corpora_v4.to_csv('data/extractions/new_data/corpora_v5.csv', index = False)

### Recuperate additional sentences 

In [7]:
corpora_v4 = pd.read_csv('data/extractions/new_data/corpora_v4.csv')

diagne_sentences = pd.read_csv('data/extractions/new_data/sentences.csv')

Let us recuperate additional sentences to augment the corpora.

- First file:

In [8]:
path = 'data/additional_documents/omniglot/sentences_french_wolof.txt'

with open(path, 'r', encoding='utf-8') as f:
    
    # recuperate lines
    omn_1 = f.readlines()
    
# recuperate sentences
omn_1_ = {
    'french': [],
    'wolof': []
}

for omn in omn_1:
    
    omn = omn.split('\t')
    
    sentences = [s.strip() for s in omn if s.strip() != '']
    
    omn_1_['french'].append(sentences[0])
    
    omn_1_['wolof'].append(sentences[1])

In [9]:
pd.options.display.max_rows = 74
omn_1_ = pd.DataFrame(omn_1_).head(74)

# print the first sentences from omniglot
omn_1_.head(74)

french  \
0                                    Bienvenue   
1                                        Salut   
2                                        Salut   
3                                        Salut   
4                             Comment vas-tu ?   
5                              Es-tu en paix ?   
6                             Comment vas-tu ?   
7                     Paix seulement, et toi ?   
8      Je suis seulement toi, comment vas-tu ?   
9      Ça fait longtemps que je ne t'ai pas vu   
10                     Comment t'appelles-tu ?   
11                     Comment t'appelles-tu ?   
12                     Comment t'appelles-tu ?   
13                             Mon nom est ...   
14                             Mon nom est ...   
15                           Je m'appelles ...   
16                             D'où viens-tu ?   
17                           D'où venez-vous ?   
18                             Je viens de ...   
19                             Je viens de ...   
20                    Plaisir de te rencontrer   
21                    Content de te rencontrer   
22            Je suis content de te rencontrer   
23                                   Bon matin   
24                            Bonne après-midi   
25                                Bonne soirée   
26                                   Au revoir   
27                                   Au revoir   
28                              Bonne chance !   
29                               Bonne santé !   
30                      Aies une bonne journée   
31                                  Bon voyage   
32                              Comprends-tu ?   
33                                Je comprends   
34                         Je ne comprends pas   
35                        Je ne le connais pas   
36                           Je ne le sais pas   
37      S'il te plaît, parles plus lentement !   
38                     Répètes s'il te plaît !   
39                       Ecris s'il te plaît !   
40                           Parles-tu anglais   
41                           Parles-tu Wolof ?   
42                                 Oui, un peu   
43            Comment dis t-on ceci en Wolof ?   
44                                Excuse-moi !   
45                          Combien ça coute ?   
46                                       Sorry   
47                                      Please   
48                                   Thank you   
49                                       Viens   
50                               Venez en paix   
51                     Où sont les toilettes ?   
52  Ce garçon peut te payer pour n'import quoi   
53             Cette jeune fille va tout payer   
54             Accepte-tu de danser avec moi ?   
55             Accepte-tu de danser avec moi ?   
56                               Tu me manques   
57                                  Je t'aimes   
58                                Je t'admires   
59                                  Je t'aimes   
60            Je te souhaites de récupérer tôt   
61                                  Vas t'en !   
62                                  Vas t'en !   
63                    Laisses moi tranquille !   
64                    Laisses moi tranquille !   
65                                  À l'aide !   
66                                       Feu !   
67                                    Arrête !   
68                         Appelle la police !   
69                           Mes voeux de noël   
70                      Mes voeux de nouvel an   
71                         Mes voeux de paques   
72                        Mes voeux de tabaski   
73                    Mes voeux d'anniversaire   

                                          wolof  
0                                        Merhbe  
1                                    Na nga def  
2                                  Na ngeen def  
3                                Salaam aleekum  
4                                 Jaam 

- Second file

In [10]:
path = 'data/additional_documents/omniglot/sentences_french_wolof_2.txt'

with open(path, 'r', encoding='utf-8') as f:
    
    # recuperate lines
    omn_2 = f.readlines()
    
# recuperate sentences
omn_2_ = {
    'french': [],
    'wolof': []
}

for omn in omn_2:
    
    omn = omn.split('–')
    
    sentences = [s.strip() for s in omn if s.strip() != '']
    
    omn_2_['wolof'].append(sentences[0])
    
    omn_2_['french'].append(sentences[1])

In [11]:
pd.options.display.max_rows = 74
omn_2_ = pd.DataFrame(omn_2_).head(74)

# print the first sentences from omniglot
omn_2_.head(74)

french                    wolof
0          Comment allez-vous ?           Na ngeen def ?
1                Bonne matinée.         Jaam nga fanane.
2                   Bonne nuit.           Fanaanal jaam.
3             À une autre fois.               Ba beneen.
4                 Je t'en prie.                   Agsil.
5              Je vous en prie.        Agsileen ak jaam.
6                       Pardon.                 Baal ma.
7                          Oui.                     Wau.
8                          Non.                 Deh-det.
9              Have you peace ?            Jaam nga am ?
10                How are you ?            Jaam nga am ?
11              Paix seulement.                Jaam rek.
12             Je suis en paix.                Jaam rek.
13                     Et toi ?                Yow nag ?
14      C'est comment ton nom ?          Naka-nga sant ?
15            D'où venez-vous ?        Fan ngeen joghe ?
16            Je viens des USA.   Maa ngi joghe les USA.
17           Parles-tu Anglais?         Deg nga Angale ?
18         Parles-tu Français ?        Deg nga Faranse ?
19  Je parle seulement anglais.    Angale rekk laa degg.
20     Je parle un peu anglais.  Degg naa tuuti Faranse.
21      Je ne parles pas Wolof.     Mahn deggumah Wolof.
22   Je ne parles pas Français.   Mahn deggumah Faranse.
23            Je n'entends pas.                 Degguma.
24               J'aimerais … .        Dama bahggoon … .
25                   Où est … ?              Fahn la … ?
26                Est-ce loin ?               Soreh na ?
27            C'est devant toi.               Cha kanam.
28                  Devant toi.               Cha kanam.
29                      Gauche.                Chammoon.
30                       Droit.                Ndeyjoor.
31                 Mais entre !           Dugghal waay !
32            Combine ça vaut ?               Lii naata?
33             C'est trop cher.           Seer na torob.

- Third file

In [12]:
path = 'data/additional_documents/omniglot/sentences_french_wolof_3.txt'

with open(path, 'r', encoding='utf-8') as f:
    
    # recuperate lines
    omn_3 = f.readlines()
    
# recuperate sentences
omn_3_ = {
    'french': [],
    'wolof': []
}

for omn in omn_3:
    
    omn = omn.split('\t')
    
    sentences = [s.strip() for s in omn if s.strip() != '']
    
    omn_3_['wolof'].append(sentences[0])
    
    omn_3_['french'].append(sentences[1])
      
        

In [13]:
pd.options.display.max_rows = 74
omn_3_ = pd.DataFrame(omn_3_).head(74)

# print the first sentences from omniglot
omn_3_.head(74)

french                             wolof
0                     Peace be upon you.                 Salaam maaleekum.
1       Que la paix soit avec toi aussi.                  Maaleekum salaam
2                        Es-tu en paix ?                     Jaama ngaam ?
3                        Paix seulement.                        Jamma rek.
4                Comment va ta famille ?                  Ana waa kër gi ?
5    Où sont les membres de ta famille ?                  Ana waa kër gi ?
6                       Ils sont là-bas.                         Ñunga fa.
7             J'espère qu'ils vont bien.                 Mbaa defuñu dara.
8               Rien ne va mal avec eux.                 Mbaa defuñu dara.
9                    Non, ils vont bien.             Déedéet, defuñu dara.
10              Rien ne va mal chez-eux.             Déedéet, defuñu dara.
11         Comment se passe le travail ?                  Naka ligéey bi ?
12             J'y suis avec, pas à pas.  Maangi ci kowam, ndanka, ndanka.
13         J'y suis avec, petit à petit.  Maangi ci kowam, ndanka, ndanka.
14                      Comment vas-tu ?                       Nanga def ?
15                      Comment vas-tu ?                         Noo def ?
16  Comment vont les gens de la maison ?                 Naka waa kër ga ?
17               Comment va la famille ?                 Naka njaboot gi ?
18                         Je vais bien.                    Maangi fi rek.
19        Qu'est-ce que tu as rapporté ?                     Lan nga indi?
20          J'ai rapporté des nouvelles.                  Xibaar laa indi.
21               Je suis dans la maison.                 Maangi ci nég bi.
22               Ils se tiennent dehors.              Ñungi taxaw ci biti.
23         As-tu passé la nuit en paix ?                Jaama nga fanaan ?
24         Comment se passe la matinée ?                   Naka subba si ?
25             La matinée se passe bien.               Subba sangi fi rek.
26      As-tu passé la matinée en paix ?                   Jaama nga endu?
27       Comment se passe l'après-midi ?                   Naka bëcëk bi ?
28           L'après-midi se passe bien.              Bëcëk baangi fi rek.
29         Comment se passes la soirée ?                   Naka ngoon si ?
30              La soirée se passe bien.              Ngoon saangi fi rek.
31            Comment se passe la nuit ?                   Naka guddi gi ?
32                La nuit se passe bien.               Guddi gaangi fi rek
33                          Comment ...?                         Noo ... ?

- Fourth file

In [14]:
path = 'data/additional_documents/hugging_face/wolof_french.csv'

hug_sents = pd.read_csv(path)

In [15]:
pd.options.display.max_rows = 74

hug_sents.head(74)

french  \
0                    J'arrive tout de suite chez toi.   
1   J'en suis sûr, cette photo ci c'est la photo p...   
2   Je vois devant moi une photo sur laquelle beau...   
3   Ceux-ci sont des personnes qui sont sortis pou...   
4   Salut ! Ceux-là qui ressemblent à des personne...   
5   Cette photo ci c'est une photo sur laquelle je...   
6   Sur la photo, ont voit des personnes qui se ré...   
7   On voit sur la photo beaucoup de personnes sor...   
8   C'est des poissons, oui. Ils sont de couleur b...   
9   Ah sur cette photo ci cependant, il y a un poi...   
10  Sur cette photo datant d'il y a trente sept an...   
11  Ça c'est nos parents Laobés qui l'ont conçu. J...   
12  Je pense que la photo que je vois là, c'est un...   
13  Ça c'est un terrain de jeu, c'est-à-dire un tr...   
14  Cette photo ci est celle d'un couteau. Un cout...   
15  Cette photo ci est une photo sur laquelle il y...   
16  Sur cette photo, il y a une chose qui ressembl...   
17  Ceci, cependant, est un bol. Il y a du sucre d...   
18  Ceci est un long et jolie bâtiment. On a vu de...   
19  Ceci est une maison. La maison est, cependant,...   
20  Sur cette photo, il y a un homme et une femme,...   
21  Sur cette photo, il y a d'une salle dans la qu...   
22  Bonjour ! Ceci est une photo. Sur cette photo ...   
23  Sur cette photo ci, il y a une calebasse. Une ...   
24  Sur cette photo, on y voit un morceau de sucre...   
25  Ceci est un lieu d'habitation très charmant, t...   
26  Oui, ceci, si tu regardes tu vois un sol sur l...   
27  Sur cette photo ci que je vois il y a des pers...   
28  Ça on dirait une place de prière... c'est-à-di...   
29  Cette photo ci que je vois c'est une photo sur...   
30  Ceci est une maison, une maison endommagée. En...   
31  Sur la photo que j'ai vue, j'y ai vu une jolie...   
32  Ceci est une théière. Une théière dans la quel...   
33  Ceci on dirait que c'est une théière, une théi...   
34  Ceci est une théière ; une théière sur la quel...   
35  Oui le parent, je vois que ceci c'est une théi...   
36  Sur la photo, que vous m'avez envoyée, il y a ...   
37  Ce que j'ai en face de moi, sur la photo, c'es...   
38  Sur la photo que j'ai, j'ai vu une sorte de ch...   
39  Cette photo ci, c'est une photo sur laquelle i...   
40  Ceci est une photo sur laquelle il y a une lar...   
41  Sur cette photo ci il y a une assurément une c...   
42  Ceci est un ancien milieu avec une constructio...   
43  Sur cette photo on voit ce qu'on appelle 'pois...   
44  Sur cette photo je suis en face de quelque cho...   
45  Cette photo ci quand je l'ai regardé j'y ai vu...   
46  Ceci est une photo sur laquelle je vois un poi...   
47  Sur la photo que j'ai eue a voir il y a assuré...   
48  Oui mon parent, je vois que ça c'est des pots ...   
49  Sur la photo, je vois des chambres avec des so...   
50  Sur la photo que j'ai eue à voir j'y ai vu des...   
51  Sur cette photo ci, j'y ai vu des chambres fai...   
52  Sur la photo que vous m'avez envoyée, j'ai vu,...   
53  Ceci est une tasse et une assiette. On y verse...   
54  Cette photo ci, c'en est une sur laquelle on e...   
55  Sur cette photo il y a des gens qui sont assis...   
56  Ceci est une chambre couverte d'herbe. En le c...   
57  Cette photo ci montre une maison qui contient ...   
58  Sur la photo que vous m'avez envoyée j'y ai vu...   
59  Sur cette photo j'y ai vu une case qui se trou...   
60  Mes salutations les plus sincères. Toujours, s...   
61  Ceux-ci sont de vieilles cases à l'intérieur d...   
62  Oui ça, il semble que ce soit une fleur, mais ...   
63  À l'intérieur de cette photo je vois beaucoup ...   
64  Je vois une large chambre avec un lit pour pas...   
65  Oui, ceci est une chambre. Une large chambre d...   
66  Sur la photo qu'on m'a montrée, c'est une cham...   
67  Sur cette photo, c'est <mask> dans laquelle on...   
68  Ceci ressemble à un bahut. Un bahut chocolat d...   
69  Ceci est un lieu d'habitation qui contient un ...   


Let us add them into the diagne's sentences and the version 4 of the corpora and remove duplicated rows.

- On corpora version 4

In [16]:
corpora_v4 = pd.concat((corpora_v4, omn_1_, omn_2_, omn_3_, hug_sents), axis=0)

# drop duplicated rows
print(sum(corpora_v4.duplicated()))

7


In [17]:
corpora_v4.drop_duplicates(inplace=True)

7 rows were deleted.

- On corpora version 5

In [18]:
diagne_sentences = pd.concat((diagne_sentences, omn_1_, omn_2_, omn_3_, hug_sents), axis=0)

# drop duplicated rows
print(sum(diagne_sentences.duplicated()))

4


In [19]:
diagne_sentences.drop_duplicates(inplace=True)

4 rows were deleted.

Let us consider only the fourth end mark adding's method and verify if we have duplicated rows.

- Corpora v4

In [21]:
french_sents = corpora_v4['french'].to_list()

wolof_sents = corpora_v4['wolof'].to_list()

french_sents = add_end_mark(french_sents, end_mark_to_remove='!')

wolof_sents = add_end_mark(wolof_sents, end_mark_to_remove='!')

corpora_v4_end = pd.DataFrame({'french': french_sents, 'wolof': wolof_sents})

In [22]:
duplicated = corpora_v4_end.duplicated()

sum(duplicated)

5

We obtain 5 duplicated rows. Let us recuperate their indexes.

In [23]:
dup_indexes = duplicated[duplicated == True].index

In [24]:
dup_indexes

Int64Index([1428, 1597, 2241, 2243, 2367], dtype='int64')

Let us remove those indexes from the corpora.

In [25]:
corpora_v4.drop(index = dup_indexes, inplace=True)

- Diagne's sentences

In [26]:
french_sents = diagne_sentences['french'].to_list()

wolof_sents = diagne_sentences['wolof'].to_list()

french_sents = add_end_mark(french_sents, end_mark_to_remove='!')

wolof_sents = add_end_mark(wolof_sents, end_mark_to_remove='!')

diagne_end = pd.DataFrame({'french': french_sents, 'wolof': wolof_sents})

In [27]:
duplicated = diagne_end.duplicated()

sum(duplicated)

5

We obtain 1 duplicated rows. Let us recuperate their indexes.

In [28]:
dup_indexes = duplicated[duplicated == True].index

In [29]:
dup_indexes

Int64Index([546, 715, 1359, 1361, 1485], dtype='int64')

Let us remove those indexes from the corpora.

In [30]:
diagne_sentences.drop(index = dup_indexes, inplace=True)

Let us save the results.

In [31]:
diagne_sentences.to_csv('data/extractions/new_data/ad_sentences.csv', index=False)

corpora_v4.to_csv('data/extractions/new_data/corpora_v6.csv',        index=False)

Let us verify how many sentences we got for the moment.

In [32]:
# from diagne's book
diagne_sentences.shape[0]

1899

In [33]:
# inside the corpora
corpora_v4.shape[0]

2781

We recuperated `1899` sentences from the diagne's book and the final corpora contains `2781` sentences.